lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +50,45 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        meta_files = sorted(self.path_to_raw_txt_data.glob('*_meta.json'),
+                            key=lambda x: get_article_id_from_filepath(x))
+        txt_files = sorted(self.path_to_raw_txt_data.glob('*_raw.txt'),
+                           key=lambda x: get_article_id_from_filepath(x))
+        if len(txt_files) != len(meta_files):
+            raise InconsistentDatasetError
+
+        for index, (meta, raw) in enumerate(zip(meta_files, txt_files), start=1):
```

Comment:
why do you pass `start=1`?

Reply:
For comparing index with get_article_id_from_filepath, the other way I will need to plus 1 to index in every iteration